ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/ReadWriteServiceBridge.scala

```diff
@@ -115,17 +115,16 @@ case class ReadWriteServiceBridge(
     //   no beginAfter supported
     //   neither multiple subscriptions
     //   neither bootstrapping the bridge from indexer persistence
-    assert(
-      beginAfter.isEmpty,
-      "Re-subscribing not supported. Only supported to subscribe once, and from inception.",
-    )
     synchronized {
       if (stateUpdatesWasCalledAlready)
         throw new IllegalStateException("not allowed to call this twice")
       else stateUpdatesWasCalledAlready = true
     }
     logger.info("Indexer subscribed to state updates.")
-    queueSource
+    beginAfter match {
```

Comment:
what is this good for...can you describe in which scenario this would help?

Reply:
I suggest you ask @kamil-da. I'm just the implementor. :upside_down_face:

Reply:
This change allows to run the `sandbox-on-x` on a database created with a Postgres export.
I needed that for testing the `ledger-api-bench-tool`. I would like to quickly prepare a running ledger with a significant amount of transactions. To do that I use existing ledger database exports.

Reply:
I'd also like to add that I don't think the previous implementation (of not conforming to the API) necessarily had a good justification. So I'm not sure why this would be _worse_.

Reply:
Thank you @nmarton-da for providing the context. 👍 
For my current work it would be sufficient just to use the patch applied locally but if you foresee value in having this change in the codebase, then I would go with the second option that @nmarton-da described - an explicit configuration parameter allowing starts with a non-empty database.

Reply:
I agree with @nmarton-da opinion that a filter sends a misleading signal. I would be fine with suggestion 1 and a disclaimer in the comments. This code is only there to facilitate our own testing at the moment.